doc/engineering/couch_views.md

```diff
@@ -1,62 +1,103 @@
-modb | Account MODB databases
-numbers | Number databases
-ratedeck | Ratedeck databases
-resource_selectors | Account resource selector databases
-system | System databases, like `system_config` or `system_media`
+`account` | The main account database
+`aggregate` | Certain databases used to aggregate documents from across accounts
+`deprecated` | Deprecated databases no longer used
+`external` | Typically the "internal" CouchDB databases for users, dbs, and nodes
+`modb` | Account MODB databases
+`numbers` | Number databases
+`ratedeck` | Ratedeck databases
+`resource_selectors` | Account resource selector databases
+`system` | System databases, like `system_config` or `system_media`
 
 Of course, applications can create their own classifications as well; they need to bind in for the database classification binding in `kzs_util` to supply the binding of their database(s).
 
-## Registering the views
+When using `classification` in `view_map`, this design document will be put on any databases that their classification is matching by this value. For our `_design/foo` example above refresh command to an account's database will put every account's views including `foo` into this database.
+
+### Database
+
+This key can be used to put the design document in specific database. In `_design/foo` example above, `"database":"system_config"` will put the design document in `system_config` database.
 
-Once the design document is updated with the `kazoo` metadata, the application will need to register the views. Generally speaking, design documents are stored in `{APP}/priv/couchdb/views/` which means the `{APP}_app` (or other init routine) would call `kz_datamgr:register_views_from_folder('{APP}', "views")`.
+## Registering The Views
+
+Each application is responsible for registering their views. Generally speaking, design documents are stored in `{APP}/priv/couchdb/views/` which means the `{APP}_app` (or other initialize routine) would call `kz_datamgr:register_views_from_folder('{APP}', "views")`. Also the application can bind to `register_view` binding, to allow system administrator to update the JSON file and re-register the view in a running node.
+
+If the application is using their own database, they have to check and create it.
+
+To put it all together, the application initialization should include some lines like below. It assumes that your application is using a database defined in `MY_DB_NAME` which defined in your application Erlang header file.
+
+```erlang
+my_app_maintenance:register_views(),
+kapps_maintenance:bind(?APP, 'my_app_maintenance', 'register_views'),
+my_app_maintenance:init_dbs(),
+```
+
+And in your `my_app_maintenance` module:
+
+```erlang
+-spec register_views() -> 'ok'.
+register_views() ->
+    _ = kz_datamgr:register_views_from_folder('{APP}', "views"),
+    'ok'.
+
+-spec init_dbs() -> 'ok'.
+init_dbs() ->
+    _ = kz_datamgr:db_create(?MY_DB_NAME),
+    _ = kz_datamgr:refresh_views(?MY_DB_NAME)
```

Comment:
you can add some logic here to only refresh_views if needed. large databases are not recommended to run this